### PR TITLE
FPLA-314: LA-Add-More-Other-Proceedings

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [X] commit messages are meaningful and follow good commit message guidelines
- [X] README and other documentation has been updated / added (if needed)
- [X] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-314

### Change description ###
Added collection 'additional proceedings' to enterProceedings case event
fixedLists in proceedings become fixedRadioLists
PDF updates to support additional proceedings
E2E and Int tests updated 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(CaseInitiationController.class)
```
Comment (brdgeth) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: Incorrect controller class.  Please change to 
@WebMvcTest(OtherProceedingsController.class)

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    @Test
+    void shouldReturnNoErrorsWhenPOnGoingProceedingIsPresent() throws Exception {
+        String onGoingProceeding = "Yes";
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(createProceeding(onGoingProceeding));
+
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
```
Comment (brdgeth) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsControllerTest.java]: Add a test for the no scenario.
String onGoingProceeding = "No";

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private static final String ERROR_MESSAGE = "You must say if there are any other proceedings relevant to this case";
+
+    @SuppressWarnings("unchecked")
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        Map<String, Object> proceedingData = (Map<String, Object>) caseDetails.getData().get("proceeding");
+        List<String> validationErrors = new ArrayList<String>();
+        String onGoingProceeding = (String)proceedingData.get("onGoingProceeding");
```
Comment (brdgeth) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/OtherProceedingsController.java]: I know its minor, however it does read better when there is a space between either side of the brackets
Change
String onGoingProceeding = (String)proceedingData.get("onGoingProceeding");
to be
String onGoingProceeding = (String) proceedingData.get("onGoingProceeding");

File: service/src/main/java/uk/gov/hmcts/reform/fpl/model/Proceeding.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import lombok.Data;
+
+@Data
+@Builder
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Proceeding {
+    private final String onGoingProceeding;
+    private final String proceedingStatus;
+    private final String caseNumber;
+    private final String started;
```
Comment (brdgeth) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/model/Proceeding.java]: Should started and ended fields be of type Date?

Comment (Jamesnelson117) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Proceeding.java]: Yes, they should! But currently in master they exist as String. If we change type we risk loosing data.

Comment (brdgeth) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/model/Proceeding.java]: Ok thats fine.

**Comment by SamanthaSJ** [comment]
PR verified - Passed

**Pull request merged**